modules/@apostrophecms/modal/ui/apos/mixins/AposModalParentMixin.js

```diff
@@ -7,15 +7,37 @@ parent components should initiate that transition on mount.
 They should also emit a `safe-close` event after the modal is completely
 closed (`modal.active` is set to `false`), at which point they themselves can
 be destroyed however necessary.
+
+If the modal parent sets `this.modified` to true, a confirmation prompt will
+appear on attempts to click cancel or use the esc key to close the modal.
+The labels can be overridden by overriding data properties as shown below.
 */
 export default {
+  data() {
```

Comment:
This all should be in the `AposEditorMixin` file since (almost) only editor modals need it. The media manager modal is the exception, which should be served via a Vue Composition API-type approach or some temporary code duplication until we're using VCA.

Reply:
Modals have other features that not all modals need. I don't think we should duplicate code or push it back to VCA over a few lines. How about it stays where it is for now, but migrates to VCA at a future date?

Reply:
I'm not saying it should be in VCA now. But this is an editor thing, so this should all be in the editor mixin. The `cancel` method should maybe look for a method to call that is in the editor mixin wrapping all the logic added in this file currently.